component/pyroscope/ebpf/ebpfspy/pprof/pprof.go

```diff
@@ -123,10 +123,10 @@ func (p *ProfileBuilder) addFunction(function string) *profile.Function {
 }
 
 func (p *ProfileBuilder) Build() ([]byte, error) {
```

Comment:
I would do `Write(io.Writer) error` instead. This way the user can deal with lifecycle of the io.Writer.

In this method you can get from the gzip writer from the pool, reset to the io.Writer received, and finally put back the writer.